gateway-ha/src/main/java/io/trino/gateway/ha/resource/GatewayWebAppResource.java

```diff
@@ -423,4 +436,67 @@ public Response readExactMatchSourceSelector()
         List<ResourceGroupsManager.ExactSelectorsDetail> selectorsDetailList = resourceGroupsManager.readExactMatchSourceSelector();
         return Response.ok(Result.ok(selectorsDetailList)).build();
     }
+
+    @GET
+    @RolesAllowed("USER")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/getRoutingRules")
+    public Response readRoutingRules()
+    {
+        String content = null;
+        try {
+            String rulesConfigPath = configuration.getRoutingRules().getRulesConfigPath();
+            content = new String(Files.readAllBytes(Paths.get(rulesConfigPath)));
+            ObjectMapper yamlReader = new ObjectMapper(new YAMLFactory());
+            YAMLParser parser = new YAMLFactory().createParser(content);
+            List<RoutingRules> routingRulesList = new ArrayList<>();
+            while (parser.nextToken() != null) {
+                RoutingRules routingRules = yamlReader.readValue(parser, RoutingRules.class);
+                routingRulesList.add(routingRules);
+            }
```

Comment:
Sure, let me try.

Reply:
I tried this but it does not seem to work. Looks like if we use this typereference code it wants the yaml file in a certain format to correctly deserialize it into an object, something like - 

`- name: "airflow"
     description: "if query from airflow, route to etl group"
     priority: 0
     condition: 'request.getHeader("X-Trino-Source") == "airflow"'
     actions:
         - 'result.put("routingGroup", "etl")' `
         
But `jeasy` rules engine is not able to read this format when we try to fetch the rules. So it doesn't seem to work.

Reply:
My mistake, I forgot `jeasy` concatenates multiple documents instead of having a list of rules. I think 

```
YAMLFactory yamlFactory = new YAMLFactory();
ObjectMapper yamlReader = new ObjectMapper(yamlFactory);

YAMLParser yamlParser = yamlFactory.createParser(rulesConfigPath)
List<ObjectNode> docs = yamlReader
      .readValues(yamlParser, new TypeReference<RoutingRulesRecord>() {})
      .readAll();
```
should work ([ref](https://stackoverflow.com/questions/25222327/deserialize-pojos-from-multiple-yaml-documents-in-a-single-file-in-jackson)).